app/controllers/verify/authentications_controller.rb

```diff
@@ -35,6 +35,12 @@ def no_auth
 
     private
 
+    def claim_parameters
```

Comment:
I guess what I was going for here was to keep the parameters coming from the `Verify::Response` super generic and only return params that related directly to the user's identity.

Reply:
Yeah, it's tricky because although that's the intent, `Verify::Response#claim_parameters` isn't really that generic. For a start the name `claim_parameters` is bound to our domain, and also we've got logic in there that is deciding which of the fields that come back are ones we care about, and also we map them to a structure that matches our domain model's attribute names, i.e. joining the names, only returning one of the names/addresses. At least by putting everything together in the same method, we own the fact that it isn't generic, and maybe at a later date we can split out the response parsing from the mapping to our model. That's something that @robbiepaul is almost certainly going to touch on with https://trello.com/c/E7ngqpON/530-fix-the-most-recent-verified-attributes.

Reply:
Ah, course. It's called `claim_parameters` after all isn't it?